cmd/prepare-storage.go

```diff
@@ -114,90 +116,199 @@ func prepForInit(disks []string, sErrs []error, diskCount int) InitActions {
+					return "online"
+				}
+				return "offline"
+			}(),
+		)
 	}
+	return msg
+}
 
-	// There is quorum or more corrupted disks, there is not enough good
-	// disks to reconstruct format.json.
-	if disksCorrupted >= quorum {
-		return Abort
+func printRegularMsg(storageDisks []StorageAPI, fn printFunc) {
+	msg := getRegularMsg(storageDisks)
+	fn(msg)
+}
+
+func getRegularMsg(storageDisks []StorageAPI) string {
+	msg := fmt.Sprintln(colorBlue("\nInitializing data volume."))
+	disksInfo, _, _ := getDisksInfo(storageDisks)
+	for i, info := range disksInfo {
+		msg += fmt.Sprintf(
+			"[%d/%d] %s - %s %s\n",
+			i+1,
+			len(storageDisks),
+			storageDisks[i],
+			humanize.IBytes(uint64(info.Total)),
+			func() string {
+				if info.Total > 0 {
+					return "online"
+				}
+				return "offline"
+			}(),
+		)
 	}
-	// Some of the formatted disks are possibly offline.
-	return WaitForHeal
+	return msg
 }
 
-func retryFormattingDisks(disks []string, storageDisks []StorageAPI) ([]StorageAPI, error) {
+func printFormatMsg(storageDisks []StorageAPI, fn printFunc) {
+	msg := getFormatMsg(storageDisks)
+	fn(msg)
+}
+
+func getFormatMsg(storageDisks []StorageAPI) string {
+	msg := fmt.Sprintln(colorBlue("\nInitializing data volume for the first time."))
+	disksInfo, _, _ := getDisksInfo(storageDisks)
+	for i, info := range disksInfo {
+		msg += fmt.Sprintf(
+			"[%d/%d] %s - %s %s\n",
+			i+1,
+			len(storageDisks),
+			storageDisks[i],
+			humanize.IBytes(uint64(info.Total)),
+			func() string {
+				if info.Total > 0 {
+					return "online"
+				}
+				return "offline"
+			}(),
+		)
+	}
+	return msg
+}
+
+func retryFormattingDisks(firstDisk bool, storageDisks []StorageAPI) error {
+	if storageDisks == nil {
+		return errInvalidArgument
+	}
 	nextBackoff := time.Duration(0)
 	var err error
-	done := false
+	var done bool
 	for !done {
 		select {
 		case <-time.After(nextBackoff * time.Second):
 			// Attempt to load all `format.json`.
 			_, sErrs := loadAllFormats(storageDisks)
-			switch prepForInit(disks, sErrs, len(storageDisks)) {
+			switch prepForInit(firstDisk, sErrs, len(storageDisks)) {
 			case Abort:
 				err = errCorruptedFormat
 				done = true
 			case FormatDisks:
+				console.Eraseline()
+				printFormatMsg(storageDisks, newPrintFn())
 				err = initFormatXL(storageDisks)
 				done = true
 			case InitObjectLayer:
+				console.Eraseline()
+				printRegularMsg(storageDisks, newPrintFn())
 				err = nil
 				done = true
+			case WaitForHeal:
+				printHealMsg(storageDisks, newPrintFn())
+			case WaitForQuorum:
+				console.Printf(
+					"Initializing data volume. Waiting for minimum %d servers to come online.\n",
+					len(storageDisks)/2+1,
+				)
+			case WaitForAll:
+				console.Println("Initializing data volume for first time. Waiting for other servers to come online.")
+			case WaitForFormatting:
+				console.Println("Initializing data volume for first time. Waiting for first server to come online.")
 			}
+			nextBackoff++
```

Comment:
Could we cycle through backoff times that are either distributed randomly within an upper-bound OR a fixed slice of suitably separated integers?